http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java

```diff
@@ -184,8 +191,14 @@ public String toString() {
 
 
     private String resolveHost() {
-        return endpoint != null ? endpoint.getHost()
-                                : resolveValue(null, ProxySystemSetting.PROXY_HOST);
+        if (endpoint != null) {
+            return endpoint.getHost();
+        }
+
+        if (Objects.equals(scheme(), "https")) {
+            return resolveValue(null, ProxySystemSetting.HTTPS_PROXY_HOST);
+        }
+        return resolveValue(null, ProxySystemSetting.PROXY_HOST);
     }
 
     private int resolvePort() {
```

Comment:
It seems `resolvePort()` is invoked _before_ `resolveScheme` in the constructor, so scheme() is always null at this point.

We could move line 66 above line 64 and pass the scheme to `resolveHost(scheme)` and `resolvePort(scheme)`

Reply:
Made changes accordingly